src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -1576,6 +1576,11 @@ private List<ActivationData> TryGetActivationDatas(List<ActivationAddress> addre
 
         public void SiloStatusChangeNotification(SiloAddress updatedSilo, SiloStatus status)
         {
+            this.ScheduleTask(() => this.OnSiloStatusChange(updatedSilo, status));
```

Comment:
Agreed, it's not great. Sergey and I settled on this as a 'good enough' fix for a patch release, but it's definitely not ideal.